FWCore/Framework/src/EventProcessor.cc

```diff
@@ -536,9 +536,30 @@ namespace edm {
     actReg_->preallocateSignal_(bounds);
     schedule_->convertCurrentProcessAlias(processConfiguration_->processName());
     pathsAndConsumesOfModules_.initialize(schedule_.get(), preg());
+    std::set<ModuleProcessName> consumedBySubProcesses;
```

Comment:
Is this variable named incorrectly? I assume `subProcess.deleteNonConsumedUnscheduledModules()` would return what was deleted, not what was kept.

Reply:
> Is this variable named incorrectly? 

The name is correct.

> I assume `subProcess.deleteNonConsumedUnscheduledModules()` would return what was deleted, not what was kept.

That function returns the set of (module label, process name) pairs that are consumed by that `subProcess` but may need to be produced by the Process.

I agree the function name could be better, I'm open for suggestions.

In principle the function could be split into two, one for fetching the consumes information from parent (Sub)Process and one for deleting the modules. On the first thought that does not feel easy because for a SubProcess having to first get the consumes information from a child SubProcess, then delete the nonconsumed modules in the current SubProcess, and then update the consumes information based on which products the current SubProcess consumes.

Reply:
Renamed the function to `keepOnlyConsumedUnscheduledModules()` (thanks to @Dr15Jones for the suggestion).